### PR TITLE
Add Hedera to react-dapp-v2 and react-wallet-v2

### PR DESCRIPTION
**Pull request by huck-hgraph**
### Summary

- Updates `dapps/react-dapp-v2` to integrate Hedera testnet
- Updates `wallets/react-wallet-v2` to integrate Hedera testnet
- Adds `@hashgraph/sdk` in dapp/wallet to build/execute native Hedera transactions

### Demo

https://github.com/hgraph-io/web-examples/assets/136644362/231db5df-f939-4db3-b35d-6c999021a0b4

#### 1. Next.config.js compiler setting

**Description:** Updates `next.config.js`, adding `compiler.styledComponents = true`.

**Rationale:** The following console warning appears whenever the app loads:
Screenshot 2023-08-07 at 11 32 26 AM
Previously, the bug only seemed to manifest visually for the `Modal` component intermittently. One consistent way to reproduce the bug is to import the `@hashgraph/sdk` lib in the `JsonRpcContext.tsx` file. Therefore, in order to properly integrate Hedera, this change to the config file is necessary.

| Before | After |
| :-: | :-: |
| Screenshot 2023-08-07 at 11 45 53 AM | Screenshot 2023-08-07 at 11 44 16 AM |

#### 2. `ScrollableData` component

**Description:** Adds new `ScrollableData` styled component that displays JSON results.
**Rationale:** The raw stringified JSON results are a bit difficult for the human eye to parse. And the way that the `Modal` component is defined, it's possible to have a payload large enough that the top and bottom of the modal render outside the viewable area. The `ScrollableData` component contains the JSON data in a more manageable sized viewing area, and formats the data to be easier to read. If the result data is just a non-JSON string, then it will display as-is.

| | Before | After |
| :-: | :-: | :-: |
| JSON | Screenshot 2023-08-07 at 11 54 13 AM | Screenshot 2023-08-07 at 11 55 28 AM |
|Regular string | Screenshot 2023-08-07 at 11 54 32 AM | Screenshot 2023-08-07 at 11 55 42 AM |

#### 3. Additional Env Vars
This PR adds the optional env vars `NEXT_PUBLIC_HEDERA_ACCOUNT_ID` and `NEXT_PUBLIC_HEDERA_PRIVATE_KEY`. These are necessary for testing the full integration of Hedera with the testnet in the demo app. However, if developers do not add these keys, it has no ill effects on the rest of the app. There is simply an error message in the console, and then everything else works as normal. Additionally, if you try to connect to a Hedera account, you will not have any accounts to approve the connection:
Screenshot 2023-08-09 at 10 30 29 AM
Screenshot 2023-08-09 at 10 38 28 AM

### CODE REVIEW

File: dapps/react-dapp-v2/src/chains/hedera.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  return metadata;
+}
+
+export function getChainRequestRender(
+  request: JsonRpcRequest
+): ChainRequestRender[] {
+  let params = [{ label: "Method", value: request.method }];
+
+  switch (request.method) {
+    default:
```
Comment (bkrem) [inline dapps/react-dapp-v2/src/chains/hedera.ts:47]: A recent PR was graceful enough to factor out the "default-only" `switch` cases where a switch wasn't actually needed, could you do the same here if it's also redundant in this case? E.g. https://github.com/WalletConnect/web-examples/blob/main/dapps/react-dapp-v2/src/chains/near.ts#L36

File: dapps/react-dapp-v2/src/chains/hedera.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  },
+};
+
+export const HederaChainData: ChainsMap = {
+  testnet: {
+    name: "Hedera Testnet",
+    id: "hedera:testnet",
+    /**
+     * Hedera integration utilizes an sdk instead of a JSON-RPC relay:
+     * {@link https://specs.walletconnect.com/2.0/blockchain-rpc/hedera-rpc}
```
Comment (bkrem) [inline dapps/react-dapp-v2/src/chains/hedera.ts:24]: This URL 404s :( Doesn't seem to be added to the specs yet?

File: dapps/react-dapp-v2/src/helpers/hedera.ts [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+const createTestnetClient = () => {
+  try {
+    /**
+     * !!!WARNING!!!
+     * Do not use this approach of storing/accessing private keys in production.
+     * This app is not configured with a secure storage mechanism, so this is just the
+     * best way to make it work and showcase Hedera integration with WalletConnect.
+     */
+    const privateKey = process.env.NEXT_PUBLIC_HEDERA_PRIVATE_KEY;
```
Comment (bkrem) [inline dapps/react-dapp-v2/src/helpers/hedera.ts:112]: I don't really understand why we need a privateKey on the dapp side, even in a demo scenario? How would this work in a dapp normally outside of any demo caveats?

Comment (bkrem) [reply dapps/react-dapp-v2/src/helpers/hedera.ts:112]: Also seeing `Failed to initialize Hedera wallet Error: Missing required env vars: ...` on the dapp which is misleading that we seem to be logging wallet errors in a sample dapp

File: wallets/react-wallet-v2/README.md [hunk 4]
```diff
@@ -22,9 +22,13 @@ Example is built atop of [NextJS](https://nextjs.org/) in order to abstract comp
 
 2. Add your project details in [WalletConnectUtil.ts](https://github.com/WalletConnect/web-examples/blob/main/wallets/react-wallet-v2/src/utils/WalletConnectUtil.ts) file
 
+3. If you want to test Hedera integration, go to [Hedera Portal](https://portal.hedera.com/) to create a Testnet account. Otherwise ignore this step.
+
 3. Install dependencies `yarn install` or `npm install`
```
Comment (bkrem) [inline wallets/react-wallet-v2/README.md:27]: Now being duplicated by `4.`, needs to be removed

```suggestion

```

File: wallets/react-wallet-v2/src/utils/HederaWalletUtil.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+     * Do not use this approach of storing/accessing private keys in production.
+     * This app is not configured with a secure storage mechanism, so this is just the
+     * best way to make it work and showcase Hedera integration with WalletConnect.
+     */
+    const privateKey = process.env.NEXT_PUBLIC_HEDERA_PRIVATE_KEY
+    const accountAddress = process.env.NEXT_PUBLIC_HEDERA_ACCOUNT_ID
+
+    if (!accountAddress || !privateKey) {
+      throw new Error(
+        'Missing required env vars: `NEXT_PUBLIC_HEDERA_ACCOUNT_ID` and/or `NEXT_PUBLIC_HEDERA_PRIVATE_KEY`'
```
Comment (bkrem) [inline wallets/react-wallet-v2/src/utils/HederaWalletUtil.ts:19]: This seems to be causing an uncaught exception when attempting to toggle Testnets on under the wallets "Settings" (gear icon) tab, which then blocks the accounts/chains view from rendering:

Screenshot 2023-08-22 at 18 38 44

**Review by bkrem — CHANGES_REQUESTED** [review]

**Comment by bkrem** [comment]
Hi @huck-hgraph,

Sorry about the delay here, I was OOO last week.

First off: really appreciate you putting in extra tweaks here around config, modal rendering etc 🙏 

I have some more general concerns around why these env vars are also required on the dapp side and found some issues that create breakage for users without the env vars set (which by default will be every dev unless they set it up).

I will do another round of deeper testing once we're a bit clearer on these aspects 👍

File: dapps/react-dapp-v2/src/chains/hedera.ts (hunk 2 above)
Comment (drgorb) [reply dapps/react-dapp-v2/src/chains/hedera.ts:24]: @bkrem won't this be added automatically once the [PR](https://github.com/WalletConnect/walletconnect-specs/pull/117) has been merged?

Comment (hgraphql) [reply dapps/react-dapp-v2/src/chains/hedera.ts]: I've updated the link to have the preview link until that PR is merged.

**Comment by hgraphql** [comment]
> Hi @huck-hgraph,
> 
> Sorry about the delay here, I was OOO last week.
> 
> First off: really appreciate you putting in extra tweaks here around config, modal rendering etc 🙏
> 
> I have some more general concerns around why these env vars are also required on the dapp side and found some issues that create breakage for users without the env vars set (which by default will be every dev unless they set it up).
> 
> I will do another round of deeper testing once we're a bit clearer on these aspects 👍

thanks @bkrem 

the env vars on the dapp side functioned as setting up some default accounts on Hedera Testnet to demonstrate using the network. For example, we need a Topic to submit a message to. I'll work on moving these set up transactions to the wallet side today. I'll also fix the other issues you've pointed out and comment back when finished.

**Comment by glitch-txs** [comment]
Hi @huck-hgraph  should we revisit this PR? can we fix the conflicts as well? Thanks

**Comment by hgraphql** [comment]
> Hi @huck-hgraph should we revisit this PR? can we fix the conflicts as well? Thanks

@glitch-txs I'll take care of this as well. Thanks!

**Pull request merged**